scipy/stats/stats.py

```diff
@@ -5670,21 +5705,145 @@ def ttest_ind(a, b, axis=0, equal_var=True, nan_policy='propagate'):
     if a.size == 0 or b.size == 0:
         return _ttest_nans(a, b, axis, Ttest_indResult)
 
-    v1 = np.var(a, axis, ddof=1)
-    v2 = np.var(b, axis, ddof=1)
-    n1 = a.shape[axis]
-    n2 = b.shape[axis]
+    if permutations is not None:
+        res = _permutation_ttest(a, b,
+                                 axis=axis,
+                                 equal_var=equal_var,
+                                 permutations=permutations,
+                                 random_state=random_state)
 
-    if equal_var:
-        df, denom = _equal_var_ttest_denom(v1, n1, v2, n2)
     else:
-        df, denom = _unequal_var_ttest_denom(v1, n1, v2, n2)
-
-    res = _ttest_ind_from_stats(np.mean(a, axis), np.mean(b, axis), denom, df)
+        v1 = np.var(a, axis, ddof=1)
+        v2 = np.var(b, axis, ddof=1)
+        n1 = a.shape[axis]
+        n2 = b.shape[axis]
 
+        if equal_var:
+            df, denom = _equal_var_ttest_denom(v1, n1, v2, n2)
+        else:
+            df, denom = _unequal_var_ttest_denom(v1, n1, v2, n2)
+        res = _ttest_ind_from_stats(np.mean(a, axis),
+                                    np.mean(b, axis),
+                                    denom, df)
     return Ttest_indResult(*res)
 
 
+def _broadcast_concatenate(xs, axis):
```

Comment:
Is there a NumPy function that can do this?
This concatenates arrays like:
```
a = [[1, 2, 3],
       [4, 5, 6]]
b = [7, 8, 9]
```
to produce
```
 [[1, 2, 3, 7, 8, 9],
  [4, 5, 6, 7, 8, 9]]
```

Reply:
Does this do what you need? 

https://numpy.org/doc/stable/reference/generated/numpy.tile.html

Reply:
Maybe there is some way you could get tile to do all this, but I'm not sure it would be any prettier.

But since I wrote this, I learned about `broadcast_to` and that might help me simplify this.

Reply:
I changed it to use `broadcast_to` instead of manually populating a pre-allocated array.